src/__internal__/input-icon-toggle/input-icon-toggle.test.tsx

```diff
@@ -0,0 +1,358 @@
+
+    const validationIcon = screen.queryByTestId(`icon-${validationProp}`);
+    const icon = screen.getByTestId("icon");
+    expect(validationIcon).not.toBeInTheDocument();
+    expect(icon).toBeVisible();
+    expect(icon).toHaveAttribute("type", "settings");
+  }
+);
+
+test.each(["error", "warning", "info"])(
+  "renders only an input icon when the validation prop is set to %s as a boolean and `useValidationIcon` is true",
+  (validationProp) => {
+    render(
+      <InputIconToggle
+        {...{
+          [validationProp]: true,
+          useValidationIcon: true,
+          inputIcon: "settings",
+        }}
+      />
+    );
+
+    const validationIcon = screen.queryByTestId(`icon-${validationProp}`);
+    const icon = screen.getByTestId("icon");
+    expect(validationIcon).not.toBeInTheDocument();
+    expect(icon).toBeVisible();
+    expect(icon).toHaveAttribute("type", "settings");
+  }
+);
+
+test.each(["error", "warning", "info"])(
+  "renders only an input icon when the validation prop is set to %s as a string and `useValidationIcon` is false",
+  (validationProp) => {
+    render(
+      <InputIconToggle
+        {...{
+          [validationProp]: "Message",
+          useValidationIcon: false,
+          inputIcon: "settings",
+        }}
+      />
+    );
+
+    const validationIcon = screen.queryByTestId(`icon-${validationProp}`);
+    const icon = screen.getByTestId("icon");
+    expect(validationIcon).not.toBeInTheDocument();
+    expect(icon).toBeVisible();
+    expect(icon).toHaveAttribute("type", "settings");
+  }
+);
+
+test.each(["error", "warning", "info"])(
+  "renders only an input icon when the validation prop is set to %s as null and `useValidationIcon` is true",
+  (validationProp) => {
+    render(
+      <InputIconToggle
+        {...{
+          [validationProp]: null,
+          useValidationIcon: true,
+          inputIcon: "settings",
+        }}
+      />
+    );
+
+    const validationIcon = screen.queryByTestId(`icon-${validationProp}`);
+    const icon = screen.getByTestId("icon");
+    expect(validationIcon).not.toBeInTheDocument();
+    expect(icon).toBeVisible();
+    expect(icon).toHaveAttribute("type", "settings");
+  }
+);
+
+test.each(["error", "warning", "info"])(
+  "renders only an input icon when the validation prop is set to %s as undefined and `useValidationIcon` is true",
+  (validationProp) => {
+    render(
+      <InputIconToggle
+        {...{
+          [validationProp]: undefined,
+          useValidationIcon: true,
+          inputIcon: "settings",
+        }}
+      />
+    );
+
+    const validationIcon = screen.queryByTestId(`icon-${validationProp}`);
+    const icon = screen.getByTestId("icon");
+    expect(validationIcon).not.toBeInTheDocument();
+    expect(icon).toBeVisible();
+    expect(icon).toHaveAttribute("type", "settings");
+  }
+);
+
+test("renders only an input icon when the `disabled` prop is true and no `validationProp` or `useValidationIcon` is set", () => {
```

Comment:
**suggestion(non-blocking)**: I was thinking about the title, and to avoid any potential confusion regarding the existence of a prop named `validationProp`, what do you think about tweaking it to something like this?:

```suggestion
test("renders an input icon, when `disabled` prop is true, and no validation prop nor `useValidationIcon` is set", () => {
```